monadic-html/src/main/scala/scala/xml/xml.scala

```diff
@@ -76,27 +76,55 @@ final case class PrefixedAttribute(
     this(pre, key, Text(value), next)
 }
 
-final case class UnprefixedAttribute(
+final case class UnprefixedAttribute[T] private (
   key: String,
   value: Node,
   next: MetaData
 ) extends MetaData {
-  def this(key: String, value: String, next: MetaData) =
-    this(key, Text(value), next)
+  def this(key: String, e: T, next: MetaData)(implicit ev: XmlSerializable[T]) =
+    this(
+      key,
+      e match {
+        case s: String => Text(s) // convenient for: attr={"computed string"}
+        case n: Node => n
+        // Iterable[Node] does not make sense in UnprefixedAttribute.
+        case _ => new Atom(e)
+      },
+      next
+    )
 }
 
-// Internal structure used by scalac to create literals -----------------------
+@scala.annotation.implicitNotFound(msg =
+  """Cannot embed value of type '${T}' into an xml literal. Allowed types are:
+Base types:
+  - String (tip: you can use ${T}.toString)
+  - Int
```

Comment:
I agree, but I'm still torn on this one because it's so convenient to not have to do `.toString` on numbers. How about we include all literal types? Int, Double, Long, Float.